fhir-search/src/main/java/com/ibm/fhir/search/SearchConstants.java

```diff
@@ -119,6 +125,12 @@ private SearchConstants() {
 
     // Extracted search parameter suffix for :text modifier
     public static final String TEXT_MODIFIER_SUFFIX = ":text";
+    
+    // Extracted search parameter suffixes for canonical values
+    public static final String CANONICAL_URL_NAME = "internal_canonical_url";
+    public static final String CANONICAL_SUFFIX = ":canonical";
```

Comment:
Since this is not a modifier, maybe we want to use something other than a colon in the suffix (like an underscore instead).

Reply:
Suffix was changed to use an underscore. CANONICAL_URL_NAME was removed. For the cases where we now create composite parameters, we use the search parameter name + CANONICAL_SUFFIX for the composite parameter name and we use the SearchUtil.makeCompositeSubCode() method for the component names. This is consistent with how other composite parameters are named.